### PR TITLE
tocpp: Support wayland

### PR DESCRIPTION
**Pull request by antonio-lunarg**
I had a look at _tocpp_, noticed there was no wayland support, so I added it. Quite an easy task thanks to [the wayland book](https://wayland-book.com/xdg-shell-basics/example-code.html).

### CODE REVIEW

**Comment by bradgrantham-lunarg** [comment]
@locke-lunarg you are welcome to look through this and I think your feedback would be helpful if you have any but I didn't want to gate this PR on your approval.

File: framework/decode/vulkan_cpp_consumer_base.cpp [hunk 1]
```diff
@@ -85,6 +85,9 @@ void VulkanCppConsumerBase::WriteMainHeader()
 {
     switch (platform_)
     {
+        default:
+            GFXRECON_LOG_FATAL("Failed to write main header: Invalid platform (%d)", platform_);
+            break;
```
Comment (locke-lunarg) [inline framework/decode/vulkan_cpp_consumer_base.cpp:90]: I barely saw `default` put on the top. It's usually on the bottom although it ran normally. What do you think? @bradgrantham-lunarg

**Comment by antonio-lunarg** [comment]
> @charles-lunarg Nothing looks amiss - easiest way to confirm if this works is to run tocpp with wayland and see if it compiles & runs.

*It works on my machine*™

It would be nice to have an automated test for this.

**Comment by locke-lunarg** [comment]
It looks good to me. If it works on your machine, that will be good, except the `default` place in switch. But I think that's ok if that's your prefer.

I wonder if you mind to add xlib support in another PR since you just added wayland support, and they might be similar. But if they are very different, please ignore what I said.

@davidlunarg I think we need ToCpp tests in our VulkanTests. That's a future plan.

**Comment by locke-lunarg** [comment]
Please add some description about Wayland support in [README.md](https://github.com/LunarG/gfxreconstruct/blob/dev/tools/tocpp/README.md) https://github.com/LunarG/gfxreconstruct/tree/dev/tools/tocpp

**Comment by antonio-lunarg** [comment]
Updated both `default:` and `README.md`.

**Review by locke-lunarg — APPROVED** [review]

**Review by bradgrantham-lunarg — APPROVED** [review]

**Pull request merged**